app/src/main/java/org/schabi/newpipe/player/helper/PlayerHelper.java

```diff
@@ -162,14 +169,45 @@ public static String resizeTypeOf(@NonNull final Context context,
 
     @NonNull
     public static String cacheKeyOf(@NonNull final StreamInfo info,
-                                    @NonNull final VideoStream video) {
-        return info.getUrl() + video.getResolution() + video.getFormat().getName();
+                                    @NonNull final VideoStream videoStream) {
+        String cacheKey = info.getUrl() + " " + videoStream.getId();
+
+        final String resolution = videoStream.getResolution();
+        final MediaFormat mediaFormat = videoStream.getFormat();
+        if (resolution.equals(RESOLUTION_UNKNOWN) && mediaFormat == null) {
+            final String content = videoStream.getContent();
+            cacheKey += " " + content.substring(20 * content.length() / 100);
```

Comment:
You forgot to change this I think ;-)
```suggestion
            cacheKey += " " + videoStream.hashCode();
```

I'd also add a comment in both `cacheKeyOf` functions explaining that the hash code is only used if the other information is not usable, since the other information is better at creating a cache key.